evap/staff/tests/test_importers.py

```diff
@@ -374,6 +387,117 @@ def test_replace_consecutive_and_trailing_spaces(self):
         success_messages, __, __ = EnrollmentImporter.process(excel_content, self.semester, None, None, test_run=True)
         self.assertIn("The import run will create 1 courses/evaluations and 3 users", "".join(success_messages))
 
+    def test_existing_course_is_not_recreated(self):
+        self.create_existing_course()
+        excel_content = excel_data.create_memory_excel_file(excel_data.test_enrollment_data_existing_course)
+
+        self.assertEqual(
+            len(
+                Course.objects.filter(
+                    semester=self.semester, name_de=self.existing_course.name_de, name_en=self.existing_course.name_en
+                )
+            ),
+            1,
+        )
+        __, warnings, __ = EnrollmentImporter.process(excel_content, self.semester, None, None, test_run=False)
+        self.assertIn(
+            "Course Existing Course (Existierender Kurs) already exists with identical attributes. Course is not created and users are put into the evaluation of that course.",
+            "".join(warnings[ImporterWarning.DUPL]),
+        )
+        self.assertEqual(
+            len(
+                Course.objects.filter(
+                    semester=self.semester, name_de=self.existing_course.name_de, name_en=self.existing_course.name_en
+                )
+            ),
+            1,
+        )
+
+    def test_existing_course_degree_is_added(self):
+        self.create_existing_course()
+        excel_content = excel_data.create_memory_excel_file(excel_data.test_enrollment_data_existing_course)
+
+        self.existing_course.degrees.set([Degree.objects.get(name_de="Master")])
+        EnrollmentImporter.process(excel_content, self.semester, None, None, test_run=False)
+        self.assertSetEqual(
+            set(self.existing_course.degrees.all()), set(Degree.objects.filter(name_de__in=["Master", "Bachelor"]))
+        )
+
+    def test_existing_course_users_added_to_evaluation(self):
+        self.create_existing_course()
+        excel_content = excel_data.create_memory_excel_file(excel_data.test_enrollment_data_existing_course)
+
+        self.assertEqual(0, len(self.existing_course.evaluations.all()[0].participants.all()))
```

Comment:
@jb3rndt You made this into `.all()[0]` instead of `.get()`, which would not throw if we had more than one evaluation. Was there a particular reason for that?

Reply:
Nope, there was no reason. It is now
`self.assertFalse(self.existing_course_evaluation.participants.exists())`